pkg/plugins/kinesis/receiver.go

```diff
@@ -371,6 +371,17 @@ func (r *Receiver) startShardReceiverEFO(svc *kinesis.Kinesis, stream *kinesis.D
 					}
 				}
 			}
+			if err := sub.EventStream.Err(); err != nil {
+				r.logger.Error().Str("op", "Kinesis.receiveWorkerEFO").Str("stream", *r.stream.StreamDescription.StreamName).Str("name", r.Name()).Str("tid", r.Tenant().ToString()).Int("shardIdx", shardIdx).Msg("subscription event stream error: " + err.Error())
+				if sub != nil && sub.EventStream != nil && sub.EventStream.Reader != nil {
+					sub.EventStream.Close()
+				}
+				time.Sleep(1 * time.Second)
```

Comment:
why we are waiting 1 second vs say 100 ms?

Reply:
Also, do we want to be more specific on the errors that we want to retry?

Reply:
It probably doesn't matter much how much we wait because under normal circumstances this line of code will never get executed, even after the subscription expires after 5 min. Rather, err will be nil and we just call line 383 sub.EventStream.Close() with no wait time at all.

Reply:
I also confirmed that our standard (non-EFO) consumer already does refresh its subscription when needed, so no bug fix is needed for that use case.

Reply:
got it, thanks!